### PR TITLE
Fix(node): delete config

### LINKED ISSUE #12677: "ghost node" (Failed to create node, unable to delete, does not exist)
```
$ minikube start --memory 4g --nodes 4 --kubernetes-version=v1.22.2
$ for i in {1..15}; do minikube node add; done
<snip>
# If you're not running a supercomputer, you'll eventually get an error from docker:

🤦  StartHost failed, but will try again: creating host: create: creating: prepare kic ssh: container name "minikube-m11" state Stopped: log: 2021-10-07T19:23:09.951028345Z  Failed to create control group inotify object: Too many open files
2021-10-07T19:23:09.951039683Z  Failed to allocate manager object: Too many open files
2021-10-07T19:23:09.951044326Z  [!!!!!!] Failed to allocate manager object.
2021-10-07T19:23:09.951049076Z  Exiting PID 1...: container exited unexpectedly
```

```
$ minikube node list 
minikube	192.168.49.2
minikube-m02	192.168.49.3
minikube-m03	192.168.49.4
minikube-m04	192.168.49.5
minikube-m05	192.168.49.6
minikube-m06	192.168.49.7
minikube-m07	192.168.49.8
minikube-m08	192.168.49.9
minikube-m09	192.168.49.10
minikube-m10	192.168.49.11
minikube-m11	
$ minikube node delete minikube-m11
🔥  Deleting node minikube-m11 from cluster minikube

❌  Exiting due to GUEST_NODE_DELETE: retrieve: Could not find node minikube-m11

╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│    Please also attach the following file to the GitHub issue:                             │
│    - /tmp/minikube_node_217bd38198816341cf01b723712436f34bbf7a1c_0.log                    │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯

$ docker ps
CONTAINER ID        IMAGE                                 COMMAND                  CREATED             STATUS              PORTS                                                                                                                                  NAMES
598aa548db9b        gcr.io/k8s-minikube/kicbase:v0.0.27   "/usr/local/bin/entr…"   25 minutes ago      Up 25 minutes       127.0.0.1:32848->22/tcp, 127.0.0.1:32847->2376/tcp, 127.0.0.1:32846->5000/tcp, 127.0.0.1:32845->8443/tcp, 127.0.0.1:32844->32443/tcp   minikube-m09
98175dfb0f18        gcr.io/k8s-minikube/kicbase:v0.0.27   "/usr/local/bin/entr…"   25 minutes ago      Up 25 minutes       127.0.0.1:32843->22/tcp, 127.0.0.1:32842->2376/tcp, 127.0.0.1:32841->5000/tcp, 127.0.0.1:32840->8443/tcp, 127.0.0.1:32839->32443/tcp   minikube-m08
f6d05b6f09fa        gcr.io/k8s-minikube/kicbase:v0.0.27   "/usr/local/bin/entr…"   26 minutes ago      Up 26 minutes       127.0.0.1:32838->22/tcp, 127.0.0.1:32837->2376/tcp, 127.0.0.1:32836->5000/tcp, 127.0.0.1:32835->8443/tcp, 127.0.0.1:32834->32443/tcp   minikube-m07
b5b53b54ed5b        gcr.io/k8s-minikube/kicbase:v0.0.27   "/usr/local/bin/entr…"   26 minutes ago      Up 26 minutes       127.0.0.1:32833->22/tcp, 127.0.0.1:32832->2376/tcp, 127.0.0.1:32831->5000/tcp, 127.0.0.1:32830->8443/tcp, 127.0.0.1:32829->32443/tcp   minikube-m06
cc2991f886c2        gcr.io/k8s-minikube/kicbase:v0.0.27   "/usr/local/bin/entr…"   27 minutes ago      Up 27 minutes       127.0.0.1:32828->22/tcp, 127.0.0.1:32827->2376/tcp, 127.0.0.1:32826->5000/tcp, 127.0.0.1:32825->8443/tcp, 127.0.0.1:32824->32443/tcp   minikube-m05
9227a06290b9        gcr.io/k8s-minikube/kicbase:v0.0.27   "/usr/local/bin/entr…"   27 minutes ago      Up 27 minutes       127.0.0.1:32823->22/tcp, 127.0.0.1:32822->2376/tcp, 127.0.0.1:32821->5000/tcp, 127.0.0.1:32820->8443/tcp, 127.0.0.1:32819->32443/tcp   minikube-m04
4a919c72ca92        gcr.io/k8s-minikube/kicbase:v0.0.27   "/usr/local/bin/entr…"   28 minutes ago      Up 28 minutes       127.0.0.1:32818->22/tcp, 127.0.0.1:32817->2376/tcp, 127.0.0.1:32816->5000/tcp, 127.0.0.1:32815->8443/tcp, 127.0.0.1:32814->32443/tcp   minikube-m03
f0f8826d1a69        gcr.io/k8s-minikube/kicbase:v0.0.27   "/usr/local/bin/entr…"   28 minutes ago      Up 28 minutes       127.0.0.1:32813->22/tcp, 127.0.0.1:32812->2376/tcp, 127.0.0.1:32811->5000/tcp, 127.0.0.1:32810->8443/tcp, 127.0.0.1:32809->32443/tcp   minikube-m02
c0fc369d5c5c        gcr.io/k8s-minikube/kicbase:v0.0.27   "/usr/local/bin/entr…"   46 minutes ago      Up 46 minutes       127.0.0.1:32788->22/tcp, 127.0.0.1:32787->2376/tcp, 127.0.0.1:32786->5000/tcp, 127.0.0.1:32785->8443/tcp, 127.0.0.1:32784->32443/tcp   minikube
$ # node 'm11' does not exist in docker, but it does in minikube and is not removeable
```

### PR DESCRIPTION
**Pull request by ayshiff**
Fixes #12677 

Resolves the "ghost node" issue.
This patch provides a way to delete a node config even if the node is not found.

I would like to add some integration tests under `TestMultiNode/DeleteNode` but I would like to have some advice on how to proceed. :)

### CODE REVIEW

File: pkg/minikube/node/node.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"github.com/pkg/errors"
 	"github.com/spf13/viper"
 
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
```
Comment (medyagh) [inline pkg/minikube/node/node.go:27]: this is not right, that error is for kuberentes not for minikube

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/node/node.go (hunk 1 above)
Comment (ayshiff) [reply pkg/minikube/node/node.go]: I thought that the error we are looking for is the one that we get in response to the `kubectl delete` Command (`client.CoreV1().Nodes().Delete` call) which is: `Error from server (NotFound): nodes "minikube-m%” not found`.

What would be the correct approach? Do we have another way to check that the node exists?   
Can we assume that if the node doesn’t have an IP address, it doesn’t exist? I tried that in my last commit.

**Comment by ayshiff** [comment]
I signed the CLA.

**Comment by mcascone** [comment]
i honestly can't tell if this is fixed or not. Is it in 1.29?

**Pull request closed without merging**